src/time.rs

```diff
@@ -101,17 +101,17 @@ impl U32Ext for u32 {
     }
 }
 
-pub fn duration(hz: Hertz, cycles: u32) -> MicroSecond {
+pub fn duration<const NOM: u32, const DENOM: u32>(hz: Hertz, cycles: u32) -> Duration<u32, NOM, DENOM> {
     let cycles = cycles as u64;
     let clk = hz.raw() as u64;
-    let us = cycles.saturating_mul(1_000_000_u64) / clk;
-    MicroSecond::from_ticks(us as u32)
+    let us = cycles.saturating_mul(DENOM as u64) / clk / NOM as u64;
```

Comment:
`let us`: Is this even microseconds any more?

Reply:
Guess not, it would be duration ticks